app/services/articles/detect_animated_images.rb

```diff
@@ -0,0 +1,38 @@
+module Articles
+  module DetectAnimatedImages
+    IMAGES_IN_LIQUID_TAGS_SELECTORS = [
+      ".liquid-comment img", # CommentTag
+      ".ltag-github-readme-tag img", # GithubReadmeTag
+      ".ltag__link__servicename img", # MediumTag
+      ".ltag__link__taglist img", # LinkTag
+      ".ltag__reddit--container img", # RedditTag
+      ".ltag__stackexchange--container img", # StackexchangeTag
+      ".ltag__twitter-tweet img", # TweetTag
+      ".ltag__user img", # UserTag and OrganizationTag
+      ".ltag__user-subscription-tag img", # UserSubscriptionTag
+      ".ltag_github-liquid-tag img", # GitHubIssueTag
+      ".podcastliquidtag img", # PodcastTag
+    ].join(", ").freeze
+
+    def self.call(article)
+      parsed_html = Nokogiri::HTML.fragment(article.processed_html)
+
+      # we ignore images contained in liquid tags as they are not animated
+      images = parsed_html.css("img") - parsed_html.css(IMAGES_IN_LIQUID_TAGS_SELECTORS)
+
+      found = false
+      images.each do |img|
+        src = img.attr("src")
+        next unless src
+
+        animated = FastImage.animated?(src)
```

Comment:
I really don't think this variable adds much in terms of readability, I'd just change the `if` below to `if FastImage.animated?(src)` Maybe something like this:

```ruby
found = false
images.each do |img|
  next unless (src = img.attr("src"))
  next unless FastImage.animated?(src)

  img["data-animated"] = true
  found = true
end
article.update_columns(processed_html: parsed_html.to_html) if found
```

Reply:
Good point! I think it was my brain trying to make operations with side effects... explicit :D